File: sdk-js/src/utils/correlation.ts
/**
 * Generate a unique correlation ID for request tracking
 */
export function generateCorrelationId(): string {
  const timestamp = Date.now()
  const random = Math.random().toString(36).substring(2, 15)
  return `req_${timestamp}_${random}`
}

/**
 * Extract correlation ID from response headers
 */
export function extractCorrelationId(headers: Headers): string | undefined {
  return headers.get('X-Correlation-Id') || undefined
}

File: sdk-py/runrgateway/utils/correlation.py
"""
Correlation ID utilities for request tracking.
"""

import time
import random
import string
from typing import Optional


def generate_correlation_id() -> str:
    """Generate a unique correlation ID for request tracking."""
    timestamp = int(time.time() * 1000)
    random_str = ''.join(random.choices(string.ascii_lowercase + string.digits, k=9))
    return f"req_{timestamp}_{random_str}"


def extract_correlation_id(headers: dict) -> Optional[str]:
    """Extract correlation ID from response headers."""
    return headers.get('X-Correlation-Id')
